src/textual/widgets/_progress_bar.py

```diff
@@ -0,0 +1,432 @@
+"""Implements a progress bar widget."""
+
+from __future__ import annotations
+
+from functools import partial
+from math import ceil
+from time import monotonic
+from typing import Optional
+
+from rich.style import Style
+
+from ..app import ComposeResult, RenderResult
+from ..containers import Horizontal
+from ..message import Message
+from ..reactive import reactive
+from ..renderables.bar import RenderableBar
+from ..timer import Timer
+from ..widget import Widget
+from ..widgets import Label
+
+
+class Bar(Widget, can_focus=False):
+    """The bar portion of the progress bar."""
+
+    COMPONENT_CLASSES = {"bar--bar", "bar--complete", "bar--indeterminate"}
+    """
+    | Class | Description |
+    | :- | :- |
+    | `bar--bar` | Style of the bar (may be used to change the color). |
+    | `bar--complete` | Style of the bar when it's complete. |
+    | `bar--indeterminate` | Style of the bar when it's in an indeterminate state. |
+    """
+
+    DEFAULT_CSS = """
+    Bar {
+        width: 32;
+        height: 1;
+    }
+    Bar > .bar--bar {
+        color: $warning;
+        background: $foreground 10%;
+    }
+    Bar > .bar--indeterminate {
+        color: $error;
+        background: $foreground 10%;
+    }
+    Bar > .bar--complete {
+        color: $success;
+        background: $foreground 10%;
+    }
+    """
+
+    _completion_percentage: reactive[float | None] = reactive[Optional[float]](None)
+    """The percentage of the bar that should be filled or `None` if indeterminate.
+
+    If the completion percentage is `None`, then the bar will render as
+    an indeterminate progress bar.
+    """
+    _refresh_timer: Timer | None
+    """Internal timer used to render the bar when it's in the indeterminate state."""
+
+    def watch__completion_percentage(self, percentage: float | None) -> None:
+        """Manage the timer that enables the indeterminate bar animation."""
+        if percentage is not None:
+            if self._refresh_timer:
+                self._refresh_timer.stop()
+                self._refresh_timer = None
+
+        else:
+            self._refresh_timer = self.set_interval(1 / 60, self.refresh)
```

Comment:
Whoah, 60FPS.  Way to high: suggest 15 FPS max for a passable animation. 

Be careful about creating things like timers. It looks like you could be creating multiple timers here.

Do you need a custom timer? Would `self.auto_refresh = 15` suffice?

Reply:
You can set `auto_refresh` to None when you don't need it to refresh. Which seems similar to what you are doing here.

Reply:
That's exactly it. On it.